sysutils/clamav-server/Portfile

```diff
@@ -3,115 +3,492 @@
-        org.macports.freshclam.plist \
-        ${LaunchDaemonsPath}
-    reinplace "s|__PREFIX__|${prefix}|g" \
-        ${LaunchDaemonsPath}/org.macports.clamd.plist \
-        ${LaunchDaemonsPath}/org.macports.freshclam.plist
     xinstall -m 0755 -o ${clamavuser} -g ${clamavuser} -d \
         ${destroot}${dataPath} \
         ${destroot}${logPath} \
-        ${destroot}${runPath}
+        ${destroot}${runPath} \
+        ${destroot}${sharePath}
     touch ${destroot}${logPath}/clamav.log
     destroot.keepdirs \
         ${destroot}${dataPath} \
         ${destroot}${runPath}
+
+    # clamd.conf.macports
+    xinstall -m 0644 -o ${clamavuser} -g ${clamavuser} ${prefix}/etc/clamd.conf.sample \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # Comment out 'Example'
+    reinplace -E {s|^(Example)[[:space:]]*$|#\1|} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # log file
+    reinplace -E {s|^[[:space:]]*#[[:space:]]*(LogFile)[[:space:]]+|\1 |} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    reinplace -E "s|^(LogFile) .+$|\\1 ${logPath}/clamd.log|" \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # LogFileMaxSize
+    reinplace -E {s|^[[:space:]]*#[[:space:]]*(LogFileMaxSize)[[:space:]]+|\1 |} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    reinplace -E "s|^(LogFileMaxSize) .+$|\\1 2M|" \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # ExtendedDetectionInfo
+    reinplace -E {s|^[[:space:]]*#[[:space:]]*(ExtendedDetectionInfo)[[:space:]]+|\1 |} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    reinplace -E "s|^(ExtendedDetectionInfo) .+$|\\1 yes|" \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # pid file
+    reinplace -E {s|^[[:space:]]*#[[:space:]]*(PidFile)[[:space:]]+|\1 |} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    reinplace -E "s|^(PidFile) .+$|\\1 ${runPath}/clamd.pid|" \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # LocalSocket
+    reinplace -E {s|^[[:space:]]*#[[:space:]]*(LocalSocket)[[:space:]]+|\1 |} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    reinplace -E "s|^(LocalSocket) .+$|\\1 ${runPath}/clamd.socket|" \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # MaxThreads
+    reinplace -E {s|^[[:space:]]*#[[:space:]]*(MaxThreads)[[:space:]]+|\1 |} \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    reinplace -E "s|^(MaxThreads) .+$|\\1 6|" \
+        ${destroot}${prefix}/etc/clamd.conf.macports
+    # ExcludePath (append after last appearance)
+    set prefix_quote_slash [strsed ${prefix} {g|/|\\/|}]
+    set preprefix_quote_slash [strsed [strsed ${prefix} {g|/[^/]*$||}] {g|/|\\/|}]
+    reinplace -E "/#ExcludePath\[^#\]*/,$!b
+//{x;//p;g;};//!H;$!d;x;s//&# BRE regex References:\\
+# https:\\/\\/www.clamav.net\\/documents\\/phishsigs#Introduction-to-regular\\
+# https:\\/\\/forum.netgate.com\\/topic\\/102819\\/alternate-definitions-for-clamav\\/10\\
```

Comment:
I suggest moving the configuration template changes into a separate file (preferably as a diff file), to avoid cluttering the Portfile.

Reply:
@l2dy Thanks. I thought about doing that, but went with specific edits instead because I believe it will be a lot easier to maintain with line-by-line edits, and more transparent to users, than re-creating patch files.

Would you accept an admittedly more-cluttered Portfile?

Reply:
Diff files provide context around the modified lines, and breaks the build if a patch failed to apply. Line-by-line edits with regex and complex escaping (making it harder to maintain) could mistakenly replace other parts of the file or miss some parts that changed in newer versions, but the port would still build so it's harder to realize that some sed commands need editing.

I think a diff file would be easier to read and understand by users not familiar with regex or sed commands, thus be more transparent than reinplace edits.

Reply:
Okay, thanks. I replaced all these Portfile line-by-line config file edits with patch files. I have to leave one in-line edit in place for a port variant specific configuration.